mithril-relay/src/relay/aggregator.rs

```diff
@@ -48,25 +51,68 @@ impl AggregatorRelay {
         }
     }
 
+    async fn notify_signer_to_aggregator(
+        &self,
+        signer_message: &RegisterSignerMessage,
+    ) -> StdResult<()> {
+        let response = reqwest::Client::new()
+            .post(format!("{}/register-signer", self.aggregator_endpoint))
+            .json(signer_message)
+            //.header(MITHRIL_API_VERSION_HEADER, "0.1.13") // TODO: retrieve current version
+            .send()
+            .await;
+        match response {
+            Ok(response) => match response.status() {
+                StatusCode::CREATED => {
+                    info!("Relay aggregator: sent successfully signer registration message to aggregator"; "signer_message" => format!("{:#?}", signer_message));
+                    Ok(())
+                }
+                status => {
+                    error!("Relay aggregator: Post `/register-signer` should have returned a 201 status code, got: {status}");
+                    Err(anyhow!("Post `/register-signer` should have returned a 201 status code, got: {status}"))
+                }
+            },
+            Err(err) => {
+                error!("Relay aggregator: Post `/register-signer` failed: {err:?}");
+                Err(anyhow!("Post `/register-signer` failed: {err:?}"))
+            }
+        }
+    }
+
     /// Tick the aggregator relay
     pub async fn tick(&mut self) -> StdResult<()> {
```

Comment:
I guess that it's going to be a nightmare to implement and produce code which is not very readbale.
We should keep it like this IMO.

Reply:
I made an attempt here: https://github.com/input-output-hk/mithril/commit/9d2113d7bdb5b6cce2f7924c0dacd9fc27830f46

But it's not successful, because of async code.

If you want to give it a try, feel free to update the branch :slightly_smiling_face: